test/rules/avoid_function_literals_in_foreach_calls.dart

```diff
@@ -12,26 +12,30 @@ void main() {
   Iterable<String> people;
 
   for (var person in people) { // OK
-    print(person);
+    print('$person!');
   }
   people.forEach((person) { // LINT
-    print(person);
+    print('$person!');
   });
 
-  people.forEach((person) => print(person)); // LINT
+  people.forEach((person) => print('$person!')); // LINT
 
   people.forEach(print); // OK
 
   people
       .where((person) => person != null)
-      .map((person) => '$person!')
-      .forEach((person) => print(person)); // OK
+      .map((person) => person.toUpperCase())
+      .forEach((person) => print('$person!')); // OK
 
   Person()
       .children
       .firstWhere((person) => person != null)
       .children
-      .forEach((person) => print(person)); // OK
+      .forEach((person) => print('$person!')); // OK
 
```

Comment:
Maybe just one more test case?

```dart
Person().children.forEach(print); // LINT
```

Reply:
Sure but did you mean "// OK" instead? that would be the case since it is using lambda just like:

`people.forEach(print); // OK`

Reply:
Added 2 more test cases for tear-offs + chaining!!

Reply:
Sorry to keep muddying the waters!

I guess what's confusing me is that I would expect this to be limited to `MethodInvocation`s based on the original proposal?

```dart
bool _hasMethodChaining(MethodInvocation node) {
  var exp = node.target;
  while (exp is PrefixedIdentifier ||
      exp is MethodInvocation ||
      exp is PropertyAccess) {
    if (exp is PrefixedIdentifier) {
      exp = (exp as PrefixedIdentifier).prefix;
    } else if (exp is MethodInvocation) {
      return true;
    } else if (exp is PropertyAccess) {
      exp = (exp as PropertyAccess).target;
    }
  }
  return false;
}
```

Reply:
Yes, but the lint is only produced if there is a closure being passed to `forEach`, and the case you suggested (`Person().children.forEach(print);`) doesn't contain one.

Reply:
Got it.  It was a bad example! 😬 

I had another confusion too I realize and was mis-reading `_hasMethodChaining` but get it now.

Thanks for bearing with me!